File: python/src/patterns/templatemethod/concrete_class_b.py
from .abstract_class import AbstractClass


class ConcreteClassB(AbstractClass):
    def primitive_operation_1(self) -> str:
        return "ConcreteClassB.primitiveOperation1"

    def primitive_operation_2(self) -> str:
        return "ConcreteClassB.primitiveOperation2"

    def hook(self) -> str:
        return " with hook"

File: typescript/src/template-method/concrete-class-b.ts
import { AbstractClass } from "./abstract-class.ts";

export class ConcreteClassB extends AbstractClass {
  protected primitiveOperation1(): string {
    return "ConcreteClassB.primitiveOperation1";
  }

  protected primitiveOperation2(): string {
    return "ConcreteClassB.primitiveOperation2";
  }

  protected override hook(): string {
    return " with hook";
  }
}
